src/nunavut/lang/cpp/support/span_lite.hpp

```diff
@@ -0,0 +1,1945 @@
+//
```

Comment:
Can we use something like [this](https://github.com/tcbrindle/span/blob/master/include/tcb/span.hpp)?

Reply:
If you insist and if Scott okays it, we could define a custom implementation of span fine-tuned for our purposes. It is hardly going to take over fifty lines of code so might be acceptable.

Reply:
The two things to consider are:

1. We cannot accept any dead code. _All_ lines of code provided by Nunavut support headers _shall_ be used by some valid DSDL type. If we have a header we want to pull in and we end up only using 10% of it we should modify the header to elide the unneeded parts.
2. Some certification regimes require exhaustive testing of all code. Any utility that makes it difficult to provide exhaustive tests on a running target is to be avoided. (We are pretty loose, in this constraint, with C++ templates but that is perhaps a tipping point where a project may want to move to C).

Reply:
I think I'll just replace both `expected` and `span` with minimalistic replacements, which are much easier to control. Maybe even throw `span` away and integrate it's functionality into `bitspan`.

Reply:
@thirtytwobits I'm pretty sure that span is going to be used throughout unless one is not going to serialize/deserialize any DSDL (we already have an option to omit serialization code completely, perhaps it should also control the emission of the span class).

@ASMfreaK can we not define `expected` simply as `using Expected = variant<Success, Error>;`? I know that it lacks the sugar but see above.

Reply:
You seem to be too keen to dig deep into bit-level optimizations. Let us please avoid that lest we end up with an assembly-style mindset where performance comes above all. I do understand the motivation but often performance-driven design comes at the cost of functional correctness: either because it nudges one to cut corners or because the implementation becomes unnecessarily complicated, potentially causing defects to creep in. What we should be focused on is an implementation that is 1. minimalistic (see rationale provided by Scott) and 2. trivial (this means that bit splitting is probably a no-go).